src/main/java/com/cloudbees/jenkins/support/configfiles/OtherConfigFilesComponent.java

```diff
@@ -99,4 +95,52 @@ public ComponentCategory getCategory() {
     }
 
     private static final Logger LOGGER = Logger.getLogger(OtherConfigFilesComponent.class.getName());
+
+    /**
+     * Extension to contribute to the list of configuration files to filter. A file need to be accepted by all
+     * {@link ConfigFilesFilter} implementations to be included in the content.
+     */
+    public interface ConfigFilesFilter extends ExtensionPoint {
+
+        /**
+         * @return all {@link ConfigFilesFilter} extensions
+         */
+        static ExtensionList<ConfigFilesFilter> all() {
+            return ExtensionList.lookup(ConfigFilesFilter.class);
+        }
+
+        static FilenameFilter getAllFilter() {
+            final List<String> filenames = all().stream()
+                    .map(ConfigFilesFilter::getFilenames)
+                    .flatMap(List::stream)
+                    .collect(Collectors.toList());
+            final Pattern allPattern = Pattern.compile("(" + String.join("|", filenames) + ")");
+            return (dir, name) -> !allPattern.matcher(name).matches();
+        }
+
+        /**
+         * Return a list of files names to exclude.
+         * @return list of file names to exclude
+         */
+        @NonNull
+        List<String> getFilenames();
```

Comment:
This API feels like a step backwards from the `FilenameFilter`. Why not something like

```java
boolean include(File f);
```

? An implementation can choose to match the file basename against a static list of names, or do something else.

Reply:
Sure. Makes sense as well.